ws-calculator/src/main/java/org/egov/wsCalculation/service/WSCalculationServiceImpl.java

```diff
@@ -0,0 +1,272 @@
+		// // this call
+		// BigDecimal collectedAmtForOldDemand =
+		// demandService.getCarryForwardAndCancelOldDemand(ptTax, criteria,
+		// requestInfo, false);
+
+		// if (collectedAmtForOldDemand.compareTo(BigDecimal.ZERO) > 0)
+		// estimates.add(TaxHeadEstimate.builder().taxHeadCode(PT_ADVANCE_CARRYFORWARD)
+		// .estimateAmount(collectedAmtForOldDemand).build());
+		// else if (collectedAmtForOldDemand.compareTo(BigDecimal.ZERO) < 0)
+		// throw new CustomException(EG_PT_DEPRECIATING_ASSESSMENT_ERROR,
+		// EG_PT_DEPRECIATING_ASSESSMENT_ERROR_MSG_ESTIMATE);
+
+		return Calculation.builder().totalAmount(totalAmount).taxAmount(taxAmt).penalty(penalty).exemption(exemption).charge(waterCharge)
+				.waterConnection(waterConnection).rebate(rebate).tenantId(tenantId).taxHeadEstimates(estimates).billingSlabIds(billingSlabIds).connectionNo(criteria.getConnectionNo()).build();
+	}
+	
+	/**
+	 * 
+	 * @param request Contains calculation request
+	 * @return List of Calculation with different tax head
+	 */
+	List<Calculation> getCalculations(CalculationReq request, Map<String, Object> masterMap) {
+		List<Calculation> calculations = new ArrayList<>(request.getCalculationCriteria().size());
+		String tenantId = request.getCalculationCriteria().get(0).getTenantId();
+		for (CalculationCriteria criteria : request.getCalculationCriteria()) {
+			Map<String, List> estimationMap = estimationService.getEstimationMap(criteria, request.getRequestInfo());
+			masterDataService.getBillingFrequencyMasterData(criteria, request.getRequestInfo(), criteria.getWaterConnection().getConnectionType(), tenantId ,masterMap);
+			Calculation calculation = getCalculation(request.getRequestInfo(), criteria, estimationMap, masterMap);
+			calculations.add(calculation);
+		}
+		return calculations;
+	}
+
+
+	@Override
+	public void jobscheduler() {
+		// TODO Auto-generated method stub
+		ArrayList<String> tenentIds = wSCalculationDao.searchTenentIds();
+
+		for (String tenentId : tenentIds) {
+			RequestInfo requestInfo = new RequestInfo();
+			User user = new User();
+			user.setTenantId(tenentId);
+			requestInfo.setUserInfo(user);
+			String jsonPath = WSCalculationConstant.JSONPATH_ROOT_FOR_BilingPeriod;
+			MdmsCriteriaReq mdmsCriteriaReq = calculatorUtil.getBillingFrequency(requestInfo, "Non Metered", tenentId);
+			StringBuilder url = calculatorUtil.getMdmsSearchUrl();
+			Object res = repository.fetchResult(url, mdmsCriteriaReq);
+			ArrayList<?> mdmsResponse = JsonPath.read(res, jsonPath);
+			if (res == null) {
+				throw new CustomException("MDMS ERROR FOR BILLING FREQUENCY",
+						"ERROR IN FETCHING THE BILLING FREQUENCY");
+			}
+			getBillingPeriod(mdmsResponse, requestInfo, tenentId);
+
+		}
+	}
+	
+
+	@SuppressWarnings("unchecked")
+	public void getBillingPeriod(ArrayList<?> mdmsResponse, RequestInfo requestInfo, String tenentId) {
+		log.info("Billing Frequency Map" + mdmsResponse.toString());
```

Comment:
This will not scale, the demand generation will have to be reworked

Reply:
ok we are pushing the final calculation response to kafka after demand generation which shall handle bulk demand generation as well